inlang/source-code/plugins/icu2-message-format/src/plugin.ts

```diff
@@ -0,0 +1,168 @@
+import type {
+	Declaration,
+	Expression,
+	FunctionAnnotation,
+	LanguageTag,
+	Message,
+	Pattern,
+	Plugin,
+	Translation,
+	Variant,
+} from "@inlang/sdk"
+import * as MF from "messageformat"
+import { displayName, description } from "../marketplace-manifest.json"
+import { PluginSettings } from "./settings.js"
+
+export const pluginId = "plugin.inlang.icu2MessageFormat"
+
+export const plugin: Plugin<{
+	[pluginId]: PluginSettings
+}> = {
+	id: pluginId,
+	displayName,
+	description,
+	settingsSchema: PluginSettings,
+	loadMessages: async ({ settings, nodeishFs }) => {
```

Comment:
we may consider adding an experimental function "importMessages" that uses the a Message2 type that can be mapped to the MessageLegacy. By doing so we can bypass the compatibility problems we face at the moment - just thinking out loud will discuss with you and @jldec tomorrow